src/components/Request.js

```diff
@@ -0,0 +1,80 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types'
+import withAuthorization from './withAuthorization'
+
+export class RequestMaker extends Component {
+    static handlersPropTypes = {
+        // Successful response handler.
+        then: PropTypes.func,
+
+        // Error handler.
+        catch: PropTypes.func,
+    }
+
+    static propTypes = {
+        // Endpoint's path.
+        from: PropTypes.string.isRequired,
+
+        // Inform about the required authorization.
+        authorized: PropTypes.bool,
+
+        // Result handlers.
+        ...Request.handlersPropTypes
+    }
+
+    static defaultProps = {
+        then: () => undefined,
+        catch: () => undefined
+    }
+
+    static get endpoint() {
+        return 'https://kudosapp-staging.herokuapp.com/'
```

Comment:
Can we have an env for it?

Reply:
Dunno. The URL was moved here as it was previously hard-coded in the (now removed) [KudosApi.js](https://github.com/Appnroll/kudosapp-frontend/blob/1e0ccb757173e228fdced7c0c9e2064183f3c0ff/src/api/KudosApi.js).